### PR TITLE
prefer c++ style casts

### PR DESCRIPTION
**Pull request by notspiff**
Use c++ style casts where cppcheck complained.
## Description
Use static/reinterpret/const casts instead of C style casts.

## Motivation and Context
Shut cppcheck up.

## How Has This Been Tested?
Compiled

### CODE REVIEW

**Comment by Rechi** [comment]
@notspiff this needs a rebase

File: xbmc/cores/VideoPlayer/DVDFileInfo.cpp [hunk 1]
```diff
@@ -454,11 +455,11 @@ bool CDVDFileInfo::DemuxerToStreamDetails(CDVDInputStream *pInputStream, CDVDDem
   // correct bluray runtime. we need the duration from the input stream, not the demuxer.
   if (pInputStream->IsStreamType(DVDSTREAM_TYPE_BLURAY))
   {
-    if(((CDVDInputStreamBluray*)pInputStream)->GetTotalTime() > 0)
+    if (static_cast<CDVDInputStreamBluray*>(pInputStream)->GetTotalTime() > 0)
     {
-      CStreamDetailVideo* detailVideo = (CStreamDetailVideo*)details.GetNthStream(CStreamDetail::VIDEO, 0);
+      CStreamDetailVideo* detailVideo = static_cast<CStreamDetailVideo*>(details.GetNthStream(CStreamDetail::VIDEO, 0));
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/DVDFileInfo.cpp]: this produces an error (cast const CStreamDetailVideo to non const CStreamDetailVideo)

Comment (notspiff) [inline xbmc/cores/VideoPlayer/DVDFileInfo.cpp]: no bluray so didn't see. fixed.

**Comment by notspiff** [comment]
rebased.

**Comment by MartijnKaijser** [comment]
rpi doesn't agree :) http://jenkins.kodi.tv/job/LINUX-RBPI/5374/console

**Comment by notspiff** [comment]
once more with more C

**Comment by Rechi** [comment]
jenkins build this please

**Comment by Rechi** [comment]
jenkins build this please

**Comment by Rechi** [comment]
jenkins build this please

**Comment by notspiff** [comment]
iirc the windows builds were the only ones which failed on last build and they both passed now.

**Comment by Rechi** [comment]
triggered a manual build and everything succeed

**Pull request merged**